slsDetectorSoftware/src/Module.cpp

```diff
@@ -3332,7 +3332,13 @@ void Module::initializeModuleStructure(detectorType type) {
     shm()->numberOfModule.y = 0;
     shm()->controlPort = DEFAULT_TCP_CNTRL_PORTNO;
     shm()->stopPort = DEFAULT_TCP_STOP_PORTNO;
-    strcpy_safe(shm()->settingsDir, getenv("HOME"));
+    char *home_directory = getenv("HOME");
+    if (home_directory != nullptr)
+        strcpy_safe(shm()->settingsDir, home_directory);
+    else {
+        strcpy_safe(shm()->settingsDir, "");
+        std::cout << "Warning, HOME directory not set" << std::endl;
```

Comment:
Could you replace this with "LOG(logWARNING) << "HOME directory not set";

So that it prints in yellow.

Reply:
@thattil please try if this is OK now.
I have no idea how to set the proper permission, in the pull request options "allow edits by maintainers" is set, but I'm no expert in Github.

Reply:
@fleon-psi All good. Ill merge it now :)